softwerkskammer/lib/groupsAndMembers/groupsAndMembersService.js

```diff
@@ -34,131 +35,123 @@ function addMembersToGroup(group, callback) {
           cb(null, members, groupsWithExtraEmailAddresses(members, groupNamesWithEmails));
         });
       }
 
-      async.reduce(groups, {}, function (memo, group, cb) {
-        groupsService.getMailinglistUsersOfList(group.id, function (err1, emails) {
+      async.reduce(groups, {}, (memo, group, cb) => {
+        groupsService.getMailinglistUsersOfList(group.id, (err1, emails) => {
           memo[group.id] = emails;
           cb(err1, memo);
         });
-      }, function (err1, groupNamesWithEmails) {
+      }, (err1, groupNamesWithEmails) => {
         loadMembersAndFillInGroups(err1, groupNamesWithEmails, callback);
       });
     });
   },
 
-  getGroupAndMembersForList: function (groupname, globalCallback) {
+  getGroupAndMembersForList: function getGroupAndMembersForList(groupname, globalCallback) {
     async.waterfall(
       [
-        function (callback) {
-          groupstore.getGroup(groupname, callback);
-        },
-        function (group, callback) {
-          addMembersToGroup(group, callback);
-        }
+        callback => groupstore.getGroup(groupname, callback),
+        (group, callback) => addMembersToGroup(group, callback)
       ],
-      function (err, group) {
-        return globalCallback(err, group);
-      }
+      globalCallback
     );
   },
 
   addMembersToGroup: addMembersToGroup,
 
-  addMembercountToGroup: function (group, callback) {
+  addMembercountToGroup: function addMembercountToGroup(group, callback) {
     if (!group) { return callback(null); }
-    groupsService.getMailinglistUsersOfList(group.id, function (err, mailinglistUsers) {
+    groupsService.getMailinglistUsersOfList(group.id, (err, mailinglistUsers) => {
       group.membercount = mailinglistUsers.length;
-      return callback(err, group);
+      callback(err, group);
     });
   },
 
-  memberIsInMemberList: function (id, members) {
-    return _(members).some(function (member) { return member.id() === id; });
+  memberIsInMemberList: function memberIsInMemberList(id, members) {
+    return members.some(member => member.id() === id);
   },
 
-  updateAdminlistSubscriptions: function (memberID, callback) {
-    this.getMemberWithHisGroupsByMemberId(memberID, function (err1, member) {
+  updateAdminlistSubscriptions: function updateAdminlistSubscriptions(memberID, callback) {
+    this.getMemberWithHisGroupsByMemberId(memberID, (err1, member) => {
       if (err1) { return callback(err1); }
-      var adminListName = conf.get('adminListName');
-      groupsService.getMailinglistUsersOfList(adminListName, function (err2, emailAddresses) {
+      const adminListName = conf.get('adminListName');
+      groupsService.getMailinglistUsersOfList(adminListName, (err2, emailAddresses) => {
         if (err2) { return callback(err2); }
-        var isInAdminList = _.includes(emailAddresses, member.email());
+        const isInAdminList = (emailAddresses || []).includes(member.email());
```

Comment:
Ich muss gestehen, das hier gefällt mir nicht so gut an der nativen Lösung. Diese Sonderfallbehandlung war vorher schön weggekapselt in der Underscore-Methode und muss jetzt an jeder Stelle wiederholt werden. Hm :-(